### PR TITLE
Document lazy-init random seed behavior in tf.random.set_seed (eager/tf.function/XLA)

### LINKED ISSUE #120627: Document Keras lazy initializer seed behavior across eager, tf.function, and XLA
### Issue type

Documentation Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

2.20.0

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 20.04

### Mobile device

_No response_

### Python version

3.9

### Bazel version

_No response_

### GCC/compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current behavior?

Keras lazy weight initialization can produce different initial weights depending on whether `call()` is executed eagerly, wrapped with `@tf.function`, or wrapped with `@tf.function(jit_compile=True)`, even when the global random seed is reset before each model definition.

This may be expected under TensorFlow's global-seed and `tf.function` random-number semantics, because no operation-level seed is specified for the layer initializer. However, the behavior is surprising in differential tests that compare eager, graph, and XLA execution modes: the outputs may differ because the models were initialized with different weights, not because compilation changed the computation.

If this behavior is expected, it would be helpful to document it more explicitly in the Keras initializer, `tf.random.set_seed`, or `tf.function` documentation, especially for models whose variables are created lazily on the first call.

### Standalone code to reproduce the issue

```python
import numpy as np
import tensorflow as tf


MODEL_CODE = """
class Model(tf.keras.Model):
    def __init__(self):
        super().__init__()
        self.d1 = tf.keras.layers.Dense(4)
        self.d2 = tf.keras.layers.Dense(3)

    def call(self, x):
        x = self.d1(x)
        return self.d2(x)


m = Model()
x = tf.constant([[1.0, 2.0, 3.0, 4.0]])
input_data = [x]
"""


def add_decorator(code, decorator):
    return code.replace("    def call", f"    {decorator}\n    def call")


def run_case(code):
    g = {"tf": tf, "np": np}
    tf.random.set_seed(42)
    np.random.seed(42)
    exec(code, g)
    g["m"](*g["input_data"])
    return g["m"].d1.kernel.numpy().copy()


w_eager = run_case(MODEL_CODE)
w_xla = run_case(add_decorator(MODEL_CODE, "@tf.function(jit_compile=True)"))
w_function = run_case(add_decorator(MODEL_CODE, "@tf.function"))

print("eager weights:", w_eager.flatten()[:4])
print("xla weights:  ", w_xla.flatten()[:4])
print("fn weights:   ", w_function.flatten()[:4])

print()
print(
    "eager == xla:",
    np.allclose(w_eager, w_xla),
    "(max diff: %.2e)" % np.abs(w_eager - w_xla).max(),
)
print(
    "eager == fn: ",
    np.allclose(w_eager, w_function),
    "(max diff: %.2e)" % np.abs(w_eager - w_function).max(),
)
print(
    "xla == fn:   ",
    np.allclose(w_xla, w_function),
    "(max diff: %.2e)" % np.abs(w_xla - w_function).max(),
)
```

### Relevant log output

```shell
eager weights: [-0.29233116  0.34708267 -0.06592196 -0.10937351]
xla weights:   [ 0.7599278  -0.0163998   0.8398861  -0.5405775 ]
fn weights:    [ 0.33809108 -0.11584264  0.12733406 -0.55019784]

eager == xla: False (max diff: 1.53e+00)
eager == fn:  False (max diff: 1.22e+00)
xla == fn:    False (max diff: 1.72e+00)
```

Expected behavior / documentation request:

If this is expected behavior, please consider documenting that resetting only the global seed may not make lazily initialized Keras weights identical across eager, `tf.function`, and `jit_compile=True` first-call paths. The recommended workaround appears to be using explicit initializer seeds, building once before comparison, or copying the same initialized weights into each execution-mode variant before differential testing.

### PR DESCRIPTION
**Pull request by Mramaran**
Fixes #120627

### What
Adds a `Note:` to the `tf.random.set_seed` docstring explaining that the global
seed only governs TensorFlow's own stateful random ops — it does not control
seeds that higher-level components (e.g. Keras layer initializers) choose for
themselves. As a result, resetting the global seed alone does not guarantee that
lazily-created variables (like layer kernels) are initialized identically across
re-runs or across eager / `tf.function` / `tf.function(jit_compile=True)`
execution. The note shows the remedy (an explicit initializer seed) and is
careful not to over-promise: an explicit seed makes eager and `tf.function`
agree, but XLA may still differ because it can lower the RNG differently.

### Why
Per #120627, this behavior is surprising in differential tests that compare
eager/graph/XLA, where the differing weights are mistaken for compilation bugs.

### Scope
Docs-only, single file: `tensorflow/python/framework/random_seed.py`. No code or
behavior change. The Keras initializer's own docs live in the external `keras`
package, so this documents the TF-side primitive.

### Verification
- Reproduced on `tf-nightly 2.22.0.dev20260625` (Keras 3): with only a global
  seed, an unseeded initializer yields different values across eager/graph/XLA;
  an explicit initializer seed aligns eager and `tf.function` (XLA may still
  differ).
- `tf_doctest` runs clean on the module; the added `python` block is illustrative
  (no `>>>`), so doctests are unaffected.

### CODE REVIEW

**Review by ratgr — APPROVED** [review]
Thank you for this documentation contribution! It clearly explains the behavior of `tf.random.set_seed` with respect to Keras layers and lazy initialization, which is a common point of confusion.
  
  No issues found. Logic and formatting look good.
